homeassistant/components/hunterdouglas_powerview/number.py

```diff
@@ -0,0 +1,124 @@
+"""Support for hunterdouglas_powerview numbers."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+from typing import Final
+
+from aiopvapi.helpers.constants import ATTR_NAME, MOTION_VELOCITY
+from aiopvapi.resources.shade import BaseShade, ShadePosition
+
+from homeassistant.components.number import (
+    NumberEntityDescription,
+    NumberMode,
+    RestoreNumber,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import PowerviewShadeUpdateCoordinator
+from .entity import ShadeEntity
+from .model import PowerviewDeviceInfo, PowerviewEntryData
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class PowerviewNumberDescriptionMixin:
+    """Mixin to describe a Number entity."""
+
+    create_entity_fn: Callable[[BaseShade], bool]
+    store_value_fn: Callable[[PowerviewShadeUpdateCoordinator, int, float | None], None]
+
+
+@dataclass(frozen=True)
```

Comment:
```suggestion
@dataclass(frozen=True, kw_only=True)
```

https://developers.home-assistant.io/blog/2023/12/11/entity-description-changes